rasa/nlu/extractors/lookup_entity_extractor.py

```diff
@@ -0,0 +1,119 @@
+import os
+import logging
+from typing import Any, Dict, List, Optional, Text
+
+import rasa.utils.common as common_utils
+from rasa.nlu.constants import ENTITIES
+from rasa.nlu.training_data import Message
+from rasa.nlu.extractors.extractor import EntityExtractor
+
+logger = logging.getLogger(__name__)
+
+
+class LookupEntityExtractor(EntityExtractor):
+    """
+    Searches for entities in the user's message from a list of examples.
+    Required Parameters:
+    @lookup -> dict
+    """
+
+    defaults = {
+        # lookup key for extracting lookup entities,
+        # it contains the dictonary of lookup entity names
+        # and their respective data files
+        # example:
+        # - name: LookupEntityExtractor
+        #   lookup:
+        #      city: /some/path/city.txt
+        #      person: /some/other/path/person.txt
+        "lookup": None
+    }
+
+    def __init__(self, component_config: Optional[Dict[Text, Any]] = None):
+        super(LookupEntityExtractor, self).__init__(component_config)
+
+        if component_config is not None and "lookup" in component_config:
+            if component_config["lookup"] is not None:
+                # check if the entities and respective file path exists
+                for key, value in list(component_config["lookup"].items()):
+                    self.add_entities_if_lookup_entity(key, value)
+            else:
+                self.component_config["lookup"] = None
+
+            if not bool(component_config["lookup"]):
+                # if the lookup dictionary is empty after filtering
+                # the lookup entities, assign it None value
+                self.component_config["lookup"] = None
+        else:
+            common_utils.raise_warning(
+                "Can't extract Lookup Entities,\
+                Please configure the lookup entities in the config.yml."
+            )
```

Comment:
I would actually raise a `ValueError` here. If no lookup dictionary is configured, the component cannot be used at all. This should result in a failure.

Reply:
Same for the other cases above. E.g. if all entries in the lookup dictionary are invalid, so that nothing is left, raise a `ValueError`.